### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.Ring.Lemmas

### PR DESCRIPTION
**Pull request by pechersky**
mathlib3 hash: 7cca171008afb30576d2d4c51173700a780c23d0

FIXME: `local notation` doesn't work, had to make just `notation` and turn off `quotPrecheck`

### CODE REVIEW

**Comment by pechersky** [comment]
@semorrison can we clear up about what the nolints file should be? I can have no newline at the end if you think that's better. Without mentioning the axioms from the new CancelMonoid, the linter won't pass.

**Comment by digama0** [comment]
The nolints file is created by `runLinter --update`, which I believe omits the final newline.

**Comment by semorrison** [comment]
It seems easier to just write the comments than to fight with the nolints file... :-)

**Comment by semorrison** [comment]
Hopefully later today a mathlib3 PR will land that renames this file to `algebra.order.ring.lemmas` (amongst other things!). Hopefully we can just follow that here before merging.

**Comment by semorrison** [comment]
@pechersky, could you please rewrite the PR description, so it only has a summary and/or todo items? Many things listed there should just be porting notes, or if they need to be actioned need zulip threads or issues.

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Tactic/Positivity/Basic.lean [hunk 1]
```diff
@@ -22,41 +23,70 @@ section Nonsense
 open Function
 -- TODO: these classes are mostly nonsense stubs which should be replaced by the real things
 -- when the theory files are ready
--- FIXME: remove this when the sorries are gone
-set_option warningAsError false
```
Comment (Ruben-VandeVelde) [inline Mathlib/Tactic/Positivity/Basic.lean:26]: :tada:

Comment (digama0) [reply Mathlib/Tactic/Positivity/Basic.lean:26]: There shouldn't be actual proofs in this file, other than one line proofs referencing other theorems and maybe rewriting a bit. These theorems presumably already exist somewhere in mathlib; please relocate them there and import them in this file.

Comment (pechersky) [reply Mathlib/Tactic/Positivity/Basic.lean:26]: Most of these proofs/instances rely on `Algebra.Order.Ring.Defs`, which in turn rely on precisely the file this PR is about. So I hope these proofs will be removed after-the-port of those files. Is that possible, or would you prefer this file is never touched at all? Basically, I'm trying to make sure that this branch has no build CI issues.

Comment (digama0) [reply Mathlib/Tactic/Positivity/Basic.lean:26]: It's fine if you have a plan to remove these proofs later. I just don't want them to be forgotten, so don't remove the `TODO` at least until then.

File: Mathlib/Algebra/Order/Ring/Lemmas.lean [hunk 2]
```diff
@@ -0,0 +1,903 @@
...
+
+/-- `MulPosStrictMono α` is an abbreviation for `CovariantClass α>0 α (λ x y, y * x) (<)`,
+expressing that multiplication by positive elements on the right is strictly monotone. -/
+abbrev MulPosStrictMono : Prop :=
+  CovariantClass α>0 α (fun x y => y * x) (· < ·)
+
+/-- `PosMulReflectLt α` is an abbreviation for `ContravariantClas α≥0 α (λ x y, x * y) (<)`,
+expressing that multiplication by nonnegative elements on the left is strictly reverse monotone. -/
+abbrev PosMulReflectLt : Prop :=
+  ContravariantClass α≥0 α (fun x y => x * y) (· < ·)
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Algebra/Order/Ring/Lemmas.lean]: Done.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**